app/services/katello/pulp/repository/yum.rb

```diff
@@ -75,19 +75,33 @@ def regenerate_applicability
           smart_proxy.pulp_api.extensions.repository.regenerate_applicability_by_ids([repo.pulp_id], true)
         end
 
+        def generate_mapping(destination_repo)
+          source_repo_map = {}
+          repo.siblings.each do |sibling|
+            key = sibling.library_instance? ? sibling.id : sibling.library_instance_id
+            source_repo_map[key] = sibling.pulp_id
+          end
+
+          Hash[destination_repo.siblings.map { |sibling| [source_repo_map[sibling.library_instance_id], sibling.pulp_id] }]
+        end
+
         def copy_contents(destination_repo, options = {})
-          override_config = Katello::Repository.build_override_config(options)
+          override_config = Katello::Repository.build_override_config(
```

Comment:
would you be open to moving the build_override_config into this service class? (since its pulp2 related).  Its only used by one more action (app/lib/actions/pulp/repository/copy_units.rb)

Reply:
you could probably have it handle injecting the mapping there as well (if we're dep solving) instead of .merge'ing it in below